src/include/socket-utils.h

```diff
@@ -159,4 +161,46 @@ sa_socklen(const struct sockaddr *sa)
         abort();
 }
 
+/**
+ * @brief Compare if two IP addresses are equal.
+ *
+ * @param sa1 The first IP address to compare.
+ *
+ * @param sa2 The second IP address to compare.
+ *
+ * @return Return 0 if they are equal and non-zero if they aren't equal.
```

Comment:
I think this needs to be adapted to say: `Returns true when equal, false otherwise.`. The next sentence can be removed.